### PR TITLE
Add swagger, getByOwner, putDog

### PR DESCRIPTION
**Pull request by haeramkeem**
Add swagger, getByOwner, putDog

### CODE REVIEW

File: src/main/java/org/cnu/realcoding/controller/DogController.java [hunk 1]
```diff
@@ -13,12 +13,22 @@
     @Autowired
     private DogManagementService dogManagementService;
 
+    @GetMapping("/dogs")
+    public List<Dog> getDogByOwnerCNTR(@PathVariable String ownerName) {
+        return dogManagementService.getDogByOwnerSRVC(ownerName);
+    }
+
+    @PutMapping
```
Comment (ywnam-stephen) [inline src/main/java/org/cnu/realcoding/controller/DogController.java]: Better add address I guess?

File: src/main/java/org/cnu/realcoding/repository/DogRepository.java [hunk 2]
```diff
@@ -66,10 +71,18 @@ public boolean check_Exist(String name, String ownerName, String ownerPhoneNumbe
...
-        criteria.is(name).and("ownerName").is(ownerName)
-                .and("ownerPhoneNumber").is(ownerPhoneNumber);
+        criteria.is(name).and("ownerName").is(ownerName).and("ownerPhoneNumber").is(ownerPhoneNumber);
         Query query = new Query(criteria);
 
         return mongoTemplate.findOne(query, Dog.class);
     }
+
+    public void putDogRPST(Dog d) {
+        Criteria crt = new Criteria("name").is(d.getName())
```
Comment (ywnam-stephen) [inline src/main/java/org/cnu/realcoding/repository/DogRepository.java]: We use name criteria for type Criteria. What a hipster.

**Review by ywnam-stephen — COMMENTED** [review]
check this

**Comment by ywnam-stephen** [comment]
K.. Guess it will be fine to merge.

File: src/main/java/org/cnu/realcoding/controller/DogController.java [hunk 3]
```diff
@@ -13,28 +13,38 @@
...
     public Dog getDogByAllKey(@PathVariable String name,
                               @PathVariable String ownerName,
                               @PathVariable String ownerPhoneNumber){
         return dogManagementService.getDogByAllKey(name, ownerName, ownerPhoneNumber);
     }
 
-    @GetMapping("/dogs/{ownerPhoneNumber}")
-    public Dog getDogByPhoneNum(@PathVariable String PhoneNum) {
+    @GetMapping
+    public Dog getDogByPhoneNum(@RequestParam String PhoneNum) {
```
Comment (kanghyejong) [inline src/main/java/org/cnu/realcoding/controller/DogController.java:39]: why do change Path Variable to RequestParam

**Pull request merged**